### PR TITLE
WIP: Issues/147 improve testing of javascript

### PR DESCRIPTION
**Pull request by cewing**
This PR implements a number of changes to testing setup and execution aimed at simplifying runtime dependency injection.

It also increases (by a small amount) the test coverage on the crew corner feature, based on these improvements.

@davisagli, @cguardia I would appreciate a review and any comments you might have on the approach.

### CODE REVIEW

File: src/vms/tests/conftest.py [hunk 1]
```diff
@@ -262,6 +210,10 @@ def leader_client(leader_environ, app, redis_client):
         }
     }))
 
+    @request.addfinalizer
+    def clean_up():
+        redis_client.hset('vms.registrations', 'cl-carla', '{}')
```
Comment (davisagli) [inline src/vms/tests/conftest.py]: Maybe it'd be better to turn the redis_client fixture into a `yield_fixture` and delete the entire vms.registrations key after yielding there?

File: src/vms/tests/factories.py [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+from faker import Faker
+import random
+
+fake = Faker()
+
+
+CONSTRUCTED_OBJECTS = {}
+
+
+class SalesforceObjectFactory(factory.BaseDictFactory):
```
Comment (davisagli) [inline src/vms/tests/factories.py]: Hmm, so now I need to learn what this mini-framework does just so I can read the tests. Why is this better than functions that return dicts?

**Comment by davisagli** [comment]
Wow, big refactor. Thanks for working on this.

As I said in my other comment I'd like to hear your reasons for using factory-boy; I'm not totally convinced.

The trick of applying mock patches as fixtures looks handy; I hadn't encountered that one.

Would you like help figuring out the python 3 failure?

File: src/vms/tests/factories.py (hunk 2 above)
Comment (cewing) [inline src/vms/tests/factories.py]: I see a few advantages over simple functions returning dictionaries:

1.  You can dynamically generate values without needing to be concerned about what they contain, so things like user names and ids and such are automatically generated.
2.  You can nest factories so that structures like a 'registration' which contain both reg data and a contact are simultaneously generated, but you can also generate nested object individually (and they will stay in sync).
3.  I use it on a number of other projects and find that it allows me to keep my data definitions confined to a smaller set, and makes it easier to keep them in sync with each-other.

Comment (davisagli) [inline src/vms/tests/factories.py]: You can do that sort of composition with plain dicts and functions too, though. i.e.

````
def contact():
    first_name = fake.first_name()
    last_name = fake.last_name()
    return {
        'first_name': first_name,
        'last_name': last_name,
        'website_member_id': o.firstname[:1].lower() + o.lastname.lower(),
    }

def registration():
    return {
        'contact': contact(),
        # etc...
    }
```

What am I missing?
